### PR TITLE
Authentication using OpenKeystore SSH API

### PR DESCRIPTION
**Pull request by vexofp**
Implementation for #71 . It seems to work well with my Yubikey 5 over NFC.

Integrating the OpenKeychain API's model (lots of PendingIntents) with the existing code (mostly modal dialog boxes) was a bit interesting. The result is not quite as elegant as I would like, but manages to avoid any intrusive refactoring of the existing authentication methods.

### CODE REVIEW

**Comment by razic** [comment]
@vexofp very nice

File: app/src/main/java/com/zeapo/pwdstore/git/GitActivity.java [hunk 1]
```diff
@@ -676,23 +681,62 @@ protected void onActivityResult(int requestCode, int resultCode,
...
+            // IdentityBuilder attempt to extract some updated state from the intent before
+            // trying to re-launch the operation.
+            if (identityBuilder != null) {
+                identityBuilder.consume(data);
+            }
+            launchGitOperation(requestCode);
         }
     }
 
+    @Override
```
Comment (MSF-Jarvis) [inline app/src/main/java/com/zeapo/pwdstore/git/GitActivity.java]: Useless override? You're not doing anything here

**Comment by zeapo** [comment]
@MSF-Jarvis we have only few tests ;(

I can start looking into this. For the moment, this needs manual testing

**Comment by zeapo** [comment]
Tested manually. It works fine. Openkeychain is surprisingly simple to use for these kind of use cases :)

**Comment by MSF-Jarvis** [comment]
> Tested manually. It works fine. Openkeychain is surprisingly simple to use for these kind of use cases :)

Sweet, I'll rebase and merge this.

**Comment by zeapo** [comment]
@vexofp I get asked to select the key on every call to jgit. Is it possible to make that selection when the user opt for OpenKeychain authentication?

Also, can you add an item in the FAQ on how to create an Authentication subkey in OpenKeychain?

**Comment by razic** [comment]
I tested this feature manually as well. It works great. Thank you very much @vexofp. To me personally, this was the feature I was missing the absolute most.

@zeapo's suggestion with regard to saving the preferred key such that you are not prompted on every git operation would definitely be a good improvement to have.

Thanks again! Looking forward to seeing this merged.

**Comment by bkchr** [comment]
Works for me as well :) Would be nice to have this into official releases!

File: app/src/main/java/com/zeapo/pwdstore/git/GitActivity.java (hunk 1 above)
Comment (vexofp) [reply app/src/main/java/com/zeapo/pwdstore/git/GitActivity.java]: Yep, this was left-over from debugging.

**Comment by vexofp** [comment]
I believe I have addressed the last few review comments and added a section to the [wiki](https://github.com/zeapo/Android-Password-Store/wiki/Generate-SSH-Key#from-openkeychain-coming-soon) on generating authentication keys in OpenKeychain, as requested by @zeapo.

I can try to work on saving the chosen key ID when I have some time, but that might be better as a separate PR.

**Comment by bkchr** [comment]
Nice :) Hopefully this gets merged then :)

**Comment by zeapo** [comment]
I'll test this tomorrow and then merge it.

Le mar. 2 avr. 2019 à 11:22, Bastian Köcher <notifications@github.com> a
écrit :

**Comment by therealpxc** [comment]
What an exciting feature!

**Comment by zeapo** [comment]
thanks :)

**Pull request merged**